project1/server/server_tasks/tport_task.cpp

```diff
@@ -0,0 +1,41 @@
+#include "tport_task.h"
+#include "../../thread_pool/thread_pool.h"
+#include "agent_task.h"
+
+#include <sys/socket.h>
+#include <cstdio>
+#include <netinet/in.h>
+#include <cstdint>
+namespace server_tasks {
+
+    thread_pool::Task::Status TPortTask::Listen() {
+        std::cout << "Server is listening for termination commands on port " << port_ << "." << std::endl;
+
+        while (true) {
+
+            // Accept a new connection.
+            int client_fd = accept(server_fd_, nullptr, nullptr);
+            if (client_fd < 0) {
+                perror("accept() failed");
+                break;
+            }
+
+            std::cout << "Handling new connection from client #" << client_fd << "." << std::endl;
+
+            auto agent_task = std::make_shared<AgentTask>();
```

Comment:
Wait, I thought the `TPortTask` was supposed to be terminating existing `Agents`, not making new ones? I'm confused...

Reply:
I've modified the `Agent` to handle `Termination Requests`. In a `TPortTask`, the `Agent` should only be handling `Termination Requests`. Inversely, the `NPortTask` will be handling all other requests.